src/MoltenFunding.sol

```diff
@@ -89,22 +99,30 @@ contract MoltenFunding is ReentrancyGuard {
         require(totalDeposited > 0, "Molten: no deposits");
 
         exchangeTime = block.timestamp;
-        exchangeRate = _exchangeRate;
+        exchangeRate = queryExchangeRate(totalDeposited);
+        assert(exchangeRate < type(uint128).max);
```

Comment:
Isn't this being done already within `queryExchangeRate`?

Reply:
good catch, thanks!